src/testdrive/src/action/kinesis/create_stream.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use rusoto_kinesis::{CreateStreamInput, DeleteStreamInput, Kinesis, ListStreamsInput};
+
+use crate::action::{Action, State};
+use crate::parser::BuiltinCommand;
+
+pub struct CreateStreamAction {
+    stream_name: String,
+}
+
+pub fn build_create_stream(mut cmd: BuiltinCommand) -> Result<CreateStreamAction, String> {
+    let stream_name = format!("testdrive-{}", cmd.args.string("stream")?);
+    cmd.args.done()?;
+
+    Ok(CreateStreamAction { stream_name })
+}
+
+impl Action for CreateStreamAction {
+    fn undo(&self, state: &mut State) -> Result<(), String> {
+        let list_streams_input = ListStreamsInput {
+            exclusive_start_stream_name: None,
+            limit: None,
+        };
+        let stream_names = match state
+            .tokio_runtime
+            .block_on(state.kinesis_client.list_streams(list_streams_input))
+        {
+            Ok(output) => output.stream_names,
+            Err(e) => {
+                return Err(format!(
+                    "hit error listing Kinesis streams: {}",
+                    e.to_string()
+                ))
+            }
+        };
```

Comment:
Testdrive style is to use `map_err` here, a la

```rust
let stream_names = state
    .tokio_runtime
    .block_on(state.kinesis_client.list_streams(list_streams_input))
    .map_err(|e| format!("listing Kinesis streams: {}", e))?
    .stream_names;
```

Reply:
Oh, two other notes!

  * `e.to_string()` is redundant; `format!` calls `to_string` on all of its arguments.

  * Proper style for error message is `format!("short context: {}", e)`. Testdrive will render errors with an `error:` prefix, so including "hit error" in the message directly is redundant.